backend/t4cclient/core/authentication/database/__init__.py

```diff
@@ -103,3 +104,17 @@ def check_username_not_in_repository(
             status_code=409,
             detail="The user already exists for this repository.",
         )
+
+
+def check_session_belongs_to_user(
+    username: str,
+    id: str,
+    db: sqlalchemy.orm.session.Session,
+):
+    sessions = get_sessions_for_user(db, username)
```

Comment:
The session does contain a username, so we can reduce the number of database calls to one. Just get the session by id and compare the username of the session.